### DISCUSSION TITLE
Why does DrawingView GetImageBytes Size have no affect?

### DISCUSSION THREAD

**Discussion by dapug** [CommunityToolkit/Maui#729] [Q&A] [answered]
I have 3 problems with DrawingView.

1. I cannot set a "canvas" size when I capture the image stream - it ALWAYS **shrinks to fit** only the space the lines actually use
2. Size seems to have NO effect in setting anything to do with the resulting stream with GetImageStream(). I can literally set it to whatever I want, same result (shrink to fit lines)
3. The resulting image (once source is set) seems to have incorrect values: Width is unchangeable (fixed at 384?), and Height changes only according to lines WIDTH (horizontal across the screen!)

```CS
        var myImage = new Image();
        var stream = await DrawingView.GetImageStream(drawView.Lines, new Size(**200,200**), Colors.Black);
        myImage.Source = ImageSource.FromStream(() => stream);
```
This results in `myImage.Height = 24` (or some number), and `myImage.Width = 384` (no matter what lines exist!)

- Note that NONE of those values had anything to do with the 200x200 control size.
- If I draw another line to the RIGHT or LEFT of the existing line, then `myImage.Height` is what changes!
- Overall, I just want the dang image to be what I told it in the Size param, and only "shrink to fit" as an option.

I'd file this as a bug but first need to be sure I understand the expected behavior.  This thing is WONKY!

**Comment by VladislavAntonyuk** [comment]
What platform do you use and do you have a repository with the reproduction sample?

**Comment by VladislavAntonyuk** [comment]
You set the size of the image in the stream. and it doesn't change the size of the Image control. It is expected behavior.
You can set Image Width and Height to 200 and DrawingView image steam size to 20 and see the effect.


As for the issue with the background. DrawingView takes "working area" of the canvas. You can get that area and put in any location you want

**Reply by dapug** [reply]
OK, if I have a 200x200 "canvas" (or area for the user to draw in, i.e. the DrawingView control that is drawn in), how do I capture and save an image that is 200x200, regardless of whether the user drew in all the available space?

By "set the size of the image in the stream", do you mean DrawingView.GetImageStream()?  I did that:
`var stream = await DrawingView.GetImageStream(drawView.Lines, new Size(200,200), Colors.Black);`

In fact, the values put in GetImageStream have NO effect, I can put anything I want in there and get the same result in the image stream.  I feel like I am missing something super simple.

Barebones sample of using DrawingView here:
https://github.com/dapug/DrawingDemo

If you draw in the center, the resulting stream size takes up exactly the dimensions of the lines, not the control:

![DV_1](https://user-images.githubusercontent.com/1814306/209642126-74ba4031-b579-4b35-a05b-3c0fc92eb9b5.png)

If you draw to the edges, you basically get an image matching the control/DrawingView area:

![DV_2](https://user-images.githubusercontent.com/1814306/209642659-235309fa-d82b-4f43-8a9a-370f3c0ecf5d.png)

Now, on one hand, it makes sense that the stream would only contain the lines and ignore/disrespect the control height/width, BUT...
1) there is no way to know what the resulting dimensions were
2) there is no way to in fact use the control dimensions as the image size when getting the stream (i.e. there is no way to specify the size in the stream because the Size param in GetImageStream() seems to do absolutely nothing.

**Reply by VladislavAntonyuk** [reply]
Your image size in stream is 200x200. 
you can save it to file and see the result.
We resize the working area and adapt it to your size (scales image by width and height - it may not be proportional). So the output image can be smaller or larger depending on your size compare to your working area. But the image size in stream is 200*200.
If I am not mistaken there are 2 methods GetImageStream. And the second one uses the same size as canvas.
You can always implement your own method to work the the image, because you have access to points/lines

**Reply by dapug** [reply]
Nope, it saves as whatever size the lines were, not the 200x200 I set in the stream and imagesource.

![image](https://user-images.githubusercontent.com/1814306/209734671-d8e2b22f-6384-4ddd-85fd-4872b6f686f9.png)
(30x23 in this case, never 200x200)

I also used the other method to get the stream, no diff.  
`var stream = await DrawingView.GetImageStream(drawView.Lines, new Size(200,200), Colors.LightBlue);`

I updated the Github repo with these changes and the file save. I added Skia because I could not figure out how to specify the file format otherwise (png in this case).  (aside: saving image files in Maui is a complete cluster nightmare compared to full .NET)

**Comment by dapug** [comment]
@VladislavAntonyuk thank you for the dialog on this. If it should work as you described, I'm thinking it warrants opening an actual issue to track.  I simply cannot get it to work like that, even in this bare bones project with clear repro.

**Comment by VladislavAntonyuk** [comment, marked as answer]
@dapug it is fixed in release 5.0.0. Now you will get a proportionally scaled image according to your desired size.

**Reply by dapug** [reply]
Thanks, that was cool that you found an issue, but I'm struggling to see different behavior.  What am I missing?  

I looked through the commit, the changes make sense.  I then updated my DrawingDemo project.  It is running v5.2.0, but I am not seeing any different behavior.  Strange, because the fix is internal to the toolkit, doesn't require any particular changes in the usage.

**Comment by ibanezjp** [comment]
Same issue as @dapug using 5.3.0

**Comment by dapug** [comment]
@VladislavAntonyuk, @ibanezjp see issue: https://github.com/CommunityToolkit/Maui/issues/1397

**Comment by CliffAgius** [comment]
Closed as answered

**Reply by JonathanGerlach** [reply]
This is answered, but it's not fixed as of version 9.1.1.

**Reply by bijington** [reply]
It is fixed in version [11.1.0](https://github.com/CommunityToolkit/Maui/releases/tag/11.1.0)
